scipy/odr/odrpack.py

```diff
@@ -171,9 +181,10 @@ class Data(object):
     Parameters
     ----------
     x : array_like
-        Input data for regression.
-    y : array_like, optional
-        Input data for regression.
+        Observed data for the independent variable of the regression
+    y : array_like or scalar, optional
```

Comment:
Yeah...that function is a little big strange to me as well, but it does seem somewhat ingrained in the codebase ATM.  Attempted changes to it caused tests to break.